app/src/main/java/org/kontalk/ui/view/LocationContentView.java

```diff
@@ -0,0 +1,176 @@
+        super(context);
+    }
+
+    public LocationContentView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public LocationContentView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+        mContent = (ImageView) findViewById(R.id.content);
+        mPlaceholder = (TextView) findViewById(R.id.placeholder);
+    }
+
+    @Override
+    public void bind(long messageId, LocationComponent component, Pattern highlight) {
+        mComponent = component;
+
+        Bitmap bitmap = getBitmap();
+        showBitmap(bitmap);
+    }
+
+    /** This method might be called from a thread other than the main thread. */
+    void showBitmap(Bitmap bitmap) {
+        // this method might be called from another thread
+        final LocationComponent component = mComponent;
+        if (component == null)
+            return;
+
+        if (bitmap != null) {
+            mContent.setImageBitmap(bitmap);
+            mPlaceholder.setVisibility(GONE);
+            mContent.setVisibility(VISIBLE);
+        }
+        else {
+            String placeholder = CompositeMessage.getSampleTextContent("Position");
+            mPlaceholder.setText(placeholder);
+            TextContentView.setTextStyle(mPlaceholder);
+            mContent.setVisibility(GONE);
+            mPlaceholder.setVisibility(VISIBLE);
+        }
+    }
+
+    private BitmapFactory.Options bitmapOptions() {
+        BitmapFactory.Options options = new BitmapFactory.Options();
+        options.inPreferredConfig = Bitmap.Config.RGB_565;
+        return options;
+    }
+
+    Bitmap loadPreview(File previewFile) throws IOException {
+        InputStream in = new FileInputStream(previewFile);
+        BitmapFactory.Options options = bitmapOptions();
+        Bitmap bitmap = BitmapFactory.decodeStream(in, null, options);
+        in.close();
+        return bitmap;
+    }
+
+    private Bitmap getBitmap() {
+
+        return BitmapFactory.decodeResource(getContext().getResources(), R.drawable.attach_location);
+    }
+
+    @Override
+    public void unbind() {
+        clear();
+    }
+
+    @Override
+    public LocationComponent getComponent() {
+        return mComponent;
+    }
+
+    /** Image is always on top. */
+    @Override
+    public int getPriority() {
+        return 1;
+    }
+
+    private void clear() {
+        mComponent = null;
+        mContent.setImageBitmap(null);
+    }
+
+    public static LocationContentView create(LayoutInflater inflater, ViewGroup parent) {
+        return (LocationContentView) inflater.inflate(R.layout.message_content_location,
+                parent, false);
+    }
+
+    interface ThumbnailListener {
+        void onThumbnailGenerated(File previewFile);
+    }
+
+    final static class GenerateThumbnailTask extends AsyncTask<Void, Void, Boolean> {
```

Comment:
I don't think this is needed here, but you've probably already replaced it with the static map loading code haven't you?